internal/mode/static/nginx/config/servers_test.go

```diff
@@ -508,44 +510,111 @@ func TestCreateServers(t *testing.T) {
 		},
 	}
 
-	expectedMatchString := func(m []httpMatch) string {
-		g := NewWithT(t)
-		b, err := json.Marshal(m)
-		g.Expect(err).ToNot(HaveOccurred())
-		return string(b)
-	}
-
-	slashMatches := []httpMatch{
-		{Method: "POST", RedirectPath: "@rule0-route0"},
-		{Method: "PATCH", RedirectPath: "@rule0-route1"},
-		{Any: true, RedirectPath: "@rule0-route2"},
-	}
-	testMatches := []httpMatch{
-		{
-			Method:       "GET",
-			Headers:      []string{"Version:V1", "test:foo", "my-header:my-value"},
-			QueryParams:  []string{"GrEat=EXAMPLE", "test=foo=bar"},
-			RedirectPath: "@rule1-route0",
+	expMatchPairs := map[string][]http.RouteMatch{
```

Comment:
we matches are the same for 8080 and 8443, we can reduce the code here. Something like below:

```
expMatchPairs := make( map[string][]http.RouteMatch)
maps.Copy(expMatchPairs, generatedExpMatchPairs("8080")
maps.Copy(expMatchPairs, generatedExpMatchPairs("8443")